### PR TITLE
Feat/language selector

### PR DESCRIPTION
**Pull request by jeffchew**
### Related Ticket(s)

https://github.com/carbon-design-system/ibm-dotcom-library/issues/2199

### Description

This adds a flag to be able to utilize a language dropdown instead of the locale button/selector. This swaps out the button with a ComboBox. The only thing missing here is changing the ComboBox direction to top, which appears to be a new feature we need to pull in when upgrading Carbon. But main functionality is all in place for this PR. 

Right now, this is behind a feature flag, so we can at least get this main functionality into the repo and try to address the combobox direction separately.

I also cleaned up a lot of the Footer docs, which looked to be very out of date.

### Changelog

**New**

- Language selector feature in the Footer

### CODE REVIEW

File: packages/styles/scss/components/footer/_language-selector.scss [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+    @include combo-box;
+    @include listbox;
+    @include text-input;
+    @include combo-box-expressive;
+    @include list-box-expressive;
+    @include text-input-expressive;
+    @include carbon--make-col-ready;
+    @include carbon--make-col(4, 4);
+
+    margin-top: carbon--mini-units(2);
```
Comment (kennylam) [inline packages/styles/scss/components/footer/_language-selector.scss:31]: Just for consistency, we've been using spacing tokens for margins/paddings

```suggestion
    margin-top: $spacing-05;
```

File: packages/react/src/components/Footer/__data__/language-items.json [hunk 2]
```diff
@@ -0,0 +1,38 @@
+[
```
Comment (kennylam) [inline packages/react/src/components/Footer/__data__/language-items.json:1]: @jeffchew Do we need to include full cc-lc for these IDs?

File: packages/react/src/components/Footer/LanguageSelector.js [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+  return (
+    <div className={`${prefix}--language-selector__container`}>
+      <ComboBox
+        id="dds-language-selector"
+        data-autoid={`${stablePrefix}--language-selector`}
+        className={`${prefix}--language-selector`}
+        onChange={({ selectedItem }) => _setSelectedItem(selectedItem)}
+        items={items}
+        itemToString={item => (item ? item.text : '')}
+        // initialSelectedItem={initialSelectedItem}
```
Comment (kennylam) [inline packages/react/src/components/Footer/LanguageSelector.js:51]: @jeffchew Did you mean to comment this out? Clearing the selected language leaves the combobox empty.

File: packages/react/src/components/Footer/__data__/language-items.json (hunk 2 above)
Comment (jeffchew) [reply packages/react/src/components/Footer/__data__/language-items.json:1]: no, these are only targeting language

File: packages/react/src/components/Footer/LanguageSelector.js (hunk 3 above)
Comment (jeffchew) [reply packages/react/src/components/Footer/LanguageSelector.js:51]: right, let me uncomment that

**Pull request merged**